packages/app/src/components/Visualization/StartScreen.js

```diff
@@ -10,18 +10,39 @@ import { withStyles } from '@material-ui/core/styles'
 import { useDataEngine } from '@dhis2/app-runtime'
 import { VisualizationError } from '../../modules/error'
 import { GenericError } from '../../assets/ErrorIcons'
+import { apiFetchVisualizations } from '../../api/visualization'
+import { visTypeIcons } from '@dhis2/analytics'
 
 const StartScreen = ({ error, classes }) => {
     const [mostViewedVisualizations, setMostViewedVisualizations] = useState([])
-
     const engine = useDataEngine()
 
     useEffect(() => {
-        async function fetchData(engine) {
-            const result = await apiFetchMostViewedVisualizations(engine, 6)
-            setMostViewedVisualizations(result.visualization)
+        async function populateMostViewedVisualizations(engine) {
+            const mostViewedVisualizationsResult = await apiFetchMostViewedVisualizations(
+                engine,
+                6
+            )
+            const visualizations = mostViewedVisualizationsResult.visualization
+            if (visualizations && visualizations.length) {
+                const visualizationsResult = await apiFetchVisualizations(
+                    engine,
+                    `id:in:[${visualizations.map(vis => vis.id)}]`,
+                    'id,type'
```

Comment:
I think this should be static in the query until it needs to be dynamic (see above), but in any case it's better to pass `fields` as an array than as a string - `['id', 'type']` so that the data engine can de-duplicate requests in the future.